master/internal/user/authz_rbac.go

```diff
@@ -243,37 +244,62 @@ func (a *UserAuthZRBAC) CanCreateAccessToken(
 	return nil
 }
 
-// CanGetAccessTokens returns an error if the user does not have permission to view either
-// their own token or another user's token based on the targetUser.
-func (a *UserAuthZRBAC) CanGetAccessTokens(
-	ctx context.Context, curUser, targetUser model.User,
-) (err error) {
+func canAdministrateAccessTokens(ctx context.Context, curUserID model.UserID) (err error) {
 	fields := audit.ExtractLogFields(ctx)
-
-	// TODO: improve logging around the case were a user is viewing their own token's description
-	if curUser.ID == targetUser.ID {
-		err = db.DoesPermissionMatch(ctx, curUser.ID, nil,
-			rbacv1.PermissionType_PERMISSION_TYPE_VIEW_TOKEN)
-		if err != nil {
-			return errors.Wrap(err, "unable to get token due to insufficient permissions")
-		}
-		return nil
-	}
-
-	logCanAdministrateAccessTokenOnUser(fields, curUser.ID,
+	logCanAdministrateAccessTokenOnUser(fields, curUserID,
 		rbacv1.PermissionType_PERMISSION_TYPE_VIEW_OTHER_TOKEN)
 	defer func() {
 		audit.LogFromErr(fields, err)
 	}()
 
-	err = db.DoesPermissionMatch(ctx, curUser.ID, nil,
+	return db.DoesPermissionMatch(ctx, curUserID, nil,
 		rbacv1.PermissionType_PERMISSION_TYPE_VIEW_OTHER_TOKEN)
-	if err != nil && curUser.ID != targetUser.ID {
+}
+
+// CanGetAccessTokens returns an error if the user does not have permission to view own or
+// another user's token based on own role permissions.
+func (a *UserAuthZRBAC) CanGetAccessTokens(
+	ctx context.Context, curUser model.User, query *bun.SelectQuery, filterUserID model.UserID,
+) (selectQuery *bun.SelectQuery, err error) {
+	err = canGetOthersAccessTokens(ctx, curUser)
+	if err != nil {
+		if filterUserID > 0 && filterUserID != curUser.ID {
+			return nil, err
+		}
+		err = canGetOwnAccessTokens(ctx, curUser)
+		if err != nil {
+			return nil, err
+		}
+		query = query.Where("us.user_id = ?", curUser.ID)
+	}
+	return query, nil
+}
+
+// CanGetOthersAccessTokens returns an error if the user does not have permission to view
+// another user's token based on own role.
+func canGetOthersAccessTokens(
+	ctx context.Context, curUser model.User,
+) (err error) {
+	err = canAdministrateAccessTokens(ctx, curUser.ID)
```

Comment:
nit (non-blocking): Would we consider instead having the `db.DoesPermissionMatch` check in here, I understand it's using the same permission as this other help function, but it makes it a little unclear what permission is needed; unless the motivation here is to make it clear that in order to view others access token `canAdministrateAccessTokens` is required across the board? 

I'm just trying to understand the trade off we're making here between having the permission check duplicated in this spot (potentially enabling us to more smoothly modify the permission check in the future if we ever decide to make it more granular instead of lumped into Administrate) vs. reducing code duplication & more strongly establishing that viewing other's tokens is an administrative operation

Reply:
I wanted to keep the` authz_rbac` and `authz_basic_impl `implementations aligned, which is why having `canAdministrateAccessTokens` made sense—since only admins currently have that capability. So, for now, this approach fits.

However, I understand your point. In the future, we might need to grant other roles the permission to view others' tokens, although I believe that requirement is unlikely to come up.

That said, I can make the logic more general by checking only the necessary permissions instead of limiting it to administrative-level permissions. Let me know what you think.